### PR TITLE
Get experiments from director

### PR DESCRIPTION
**Pull request by dmitryagapov**
- [x] Get experiments list
- [x] Get experiment
  - [x] Get tasks
  - [x] Get downloads status
  - [x] Get collaborators

### CODE REVIEW

**Review by igor-davidyuk — COMMENTED** [review]
Could we also restore the `FLExperiment` object in a jupyter notebook from the information on the Director?
Imagine that we restarted the notebook and want to get the object back, or we want to manage some experiment from another notebook or client.

File: openfl/protocols/director.proto [hunk 1]
```diff
@@ -10,128 +10,186 @@ import "federation.proto";
...
+message DownloadStatuses {
+  repeated DownloadStatus models = 1;
+  repeated DownloadStatus logs = 2;
+}
+
+message GetExperimentDescriptionResponse {
+  string name = 1;
+  string status = 2;
+  float progress = 3;
+  uint32 totalRounds = 4;
```
Comment (aleksandr-mokrov) [inline openfl/protocols/director.proto]: `totalRounds` -> `total_rounds`

File: openfl/protocols/director.proto [hunk 2]
```diff
@@ -10,128 +10,186 @@ import "federation.proto";
...
+  uint32 currentRound = 5;
```
Comment (aleksandr-mokrov) [inline openfl/protocols/director.proto]: `currentRound` -> `current_round`

File: openfl/protocols/director.proto [hunk 3]
```diff
@@ -10,128 +10,186 @@ import "federation.proto";
...
+  DownloadStatuses downloadStatuses = 6;
```
Comment (aleksandr-mokrov) [inline openfl/protocols/director.proto]: `downloadStatuses` -> `download_statuses`

File: openfl/protocols/director.proto [hunk 4]
```diff
@@ -10,128 +10,186 @@ import "federation.proto";
...
+message GetExperimentDescriptionRequest {
+  string name = 1;
+}
+
+message CollaboratorDescription {
+  string name = 1;
+  string status = 2;
+  float progress = 3;
+  uint32 round = 4;
+  string currentTask = 5;
```
Comment (aleksandr-mokrov) [inline openfl/protocols/director.proto]: `currentTask` -> `current_task`

File: openfl/protocols/director.proto [hunk 5]
```diff
@@ -10,128 +10,186 @@ import "federation.proto";
...
+  string nextTask = 6;
```
Comment (aleksandr-mokrov) [inline openfl/protocols/director.proto]: `nextTask` -> `next_task`

File: openfl/component/director/experiment.py [hunk 6]
```diff
@@ -21,10 +21,10 @@
 class Status:
     """Experiment's statuses."""
 
-    PENDING = 'PENDING'
-    FINISHED = 'FINISHED'
-    IN_PROGRESS = 'IN_PROGRESS'
-    FAILED = 'FAILED'
+    PENDING = 'pending'
+    FINISHED = 'finished'
+    IN_PROGRESS = 'inProgress'
```
Comment (aleksandr-mokrov) [inline openfl/component/director/experiment.py]: `inProgress` -> `in_progress`

File: openfl/protocols/director.proto [hunk 7]
```diff
@@ -10,128 +10,186 @@ import "federation.proto";
...
+  string status = 2;
+}
+
+message DownloadStatuses {
+  repeated DownloadStatus models = 1;
+  repeated DownloadStatus logs = 2;
+}
+
+message GetExperimentDescriptionResponse {
+  string name = 1;
```
Comment (aleksandr-mokrov) [inline openfl/protocols/director.proto]: Why does the response consists of separate experiment's fields, not of `ExperimentDescription`?

**Review by aleksandr-mokrov — CHANGES_REQUESTED** [review]

File: openfl/protocols/director.proto (hunk 1 above)
Comment (dmitryagapov) [reply openfl/protocols/director.proto]: Done

File: openfl/protocols/director.proto (hunk 2 above)
Comment (dmitryagapov) [reply openfl/protocols/director.proto]: Done

File: openfl/protocols/director.proto (hunk 3 above)
Comment (dmitryagapov) [reply openfl/protocols/director.proto]: Done

File: openfl/protocols/director.proto (hunk 5 above)
Comment (dmitryagapov) [reply openfl/protocols/director.proto]: Done

File: openfl/component/director/experiment.py (hunk 6 above)
Comment (dmitryagapov) [reply openfl/component/director/experiment.py]: Done

Comment (dmitryagapov) [reply openfl/component/director/experiment.py]: Done

File: openfl/protocols/director.proto (hunk 7 above)
Comment (dmitryagapov) [reply openfl/protocols/director.proto]: Done

**Review by aleksandr-mokrov — APPROVED** [review]

File: openfl/component/director/director.py [hunk 8]
```diff
@@ -192,12 +198,63 @@ def get_envoys(self) -> list:
...
-                time.time() < envoy_info['last_updated']
-                + envoy_info['valid_duration']
+                time.time() < envoy_info['last_updated'] + envoy_info['valid_duration']
             )
+            envoy_name = envoy_info['shard_info']['node_info']['name']
+            envoy_info['experiment_name'] = self.col_exp[envoy_name]
 
         return self._shard_registry.values()
 
+    def get_experiments(self, caller: str) -> list:
```
Comment (IgorDavidyuk) [inline openfl/component/director/director.py:208]: We have a naming collapse here. 
To the existing `get_experiment_data` method we add `get_experiments` and `get_experiment`.
Could we resolve methods naming and merge some methods if possible?

Comment (igor-davidyuk) [reply openfl/component/director/director.py:208]: +1

File: openfl/component/director/director.py [hunk 9]
```diff
@@ -192,12 +198,63 @@ def get_envoys(self) -> list:
         logger.info(f'Shard registry: {self._shard_registry}')
```
Comment (igor-davidyuk) [inline openfl/component/director/director.py:198]: We also have `get_envoys` and `get_envoys` that are virtually synonyms

File: openfl/protocols/director.proto [hunk 10]
```diff
@@ -133,6 +191,10 @@ service FederationDirector {
     rpc GetExperimentData (GetExperimentDataRequest) returns (stream ExperimentData) {}
     rpc EnvoyHealthCheck (EnvoyStatus) returns (EnvoyHealthCheckResponse) {}
 
+    // Experiments RPCs
```
Comment (igor-davidyuk) [inline openfl/protocols/director.proto:196]: These RPCs are also made by front-end clients. In this sense, those are also API RPCs (better rename to FRONTEND RPCs)

File: openfl/plugins/processing_units_monitor/pynvml_monitor.py [hunk 11]
```diff
@@ -3,24 +3,25 @@
 """
 pynvml CUDA Device monitor plugin module.
 
-Required package: nvidia-ml-py3
+Required package: pynvml
```
Comment (igor-davidyuk) [inline openfl/plugins/processing_units_monitor/pynvml_monitor.py:6]: You cat create just one additional plugin that uses `pynvml`. No need to remove the old one.

**Review by igor-davidyuk — APPROVED** [review]

File: openfl/plugins/processing_units_monitor/pynvml_monitor.py (hunk 11 above)
Comment (dmitryagapov) [reply openfl/plugins/processing_units_monitor/pynvml_monitor.py:6]: `nvidia-ml-py3` looks like the old version of `pynvml`.
For `nvidia-ml-py3` last commit was 3 years ago.

File: openfl/protocols/director.proto (hunk 10 above)
Comment (dmitryagapov) [reply openfl/protocols/director.proto:196]: Separate issue was created https://github.com/intel/openfl/issues/222

**Pull request merged**